src/Microsoft.Security.Utilities.Core/CachedDotNetRegex.cs

```
@@ -44,32 +44,91 @@ public IEnumerable<UniversalMatch> Matches(string input, string pattern, RegexOp
             Regex regex = GetOrCreateRegex(pattern, options);
             foreach (Match m in regex.Matches(input))
             {
-                yield return ToFlex(m, captureGroup);
+                yield return CreateUniversalMatch(captureGroup: captureGroup, match: m);
 
                 // Instance Regex.Matches has no overload; check timeout between matches
-                // (MatchesCollection *is* lazily computed).
+                // (MatchCollection *is* lazily computed).
```

I'm not loving this timeout. There's no way to pass it all the way to here when using SecretMasker. Also, it's only *between* regexes and I find silently dropping remaining matches vs. throw surprising. Also RE2 engine (in our tests) doesn't use it in the code path with refine group (the only one we use). I thought about ripping out timeout, but decided against unrelated breaking change on top of the related ones. :) #Closed

---

Separate and I'm not sure it's worth an issue.

---

Discussion offline concluded we should file an issue. So I did: #229